packages/statemanager/src/ethersStateManager.ts

```diff
@@ -0,0 +1,400 @@
+   * Cannot be more than 32 bytes. Leading zeros are stripped.
+   * If it is empty or filled with zeros, deletes the value.
+   */
+  async putContractStorage(address: Address, key: Buffer, value: Buffer): Promise<void> {
+    const storageTrie = await this._getStorageTrie(address)
+    if (value.length > 0)
+      await storageTrie.put(
+        key,
+        arrToBufArr(RLP.encode(value.length > 0 ? value : Buffer.from([])))
+      )
+    else await storageTrie.del(key)
+
+    const contract = await this.getAccount(address)
+    contract.storageRoot = storageTrie.root()
+
+    await this.putAccount(address, contract)
+  }
+
+  /**
+   * Clears all storage entries for the account corresponding to `address`.
+   * @param address -  Address to clear the storage of
+   */
+  async clearContractStorage(address: Address): Promise<void> {
+    const storageTrie = await this._getStorageTrie(address)
+    storageTrie.root(this.trie.EMPTY_TRIE_ROOT)
+    const contract = await this.getAccount(address)
+    contract.storageRoot = storageTrie.root()
+    await this.putAccount(address, contract)
+  }
+
+  /**
+   * Dumps the RLP-encoded storage values for an `account` specified by `address`.
+   * @param address - The address of the `account` to return storage for
+   * @returns {Promise<StorageDump>} - The state of the account as an `Object` map.
+   * Keys are are the storage keys, values are the storage values as strings.
+   * Both are represented as `0x` prefixed hex strings.
+   */
+  dumpStorage(address: Address): Promise<StorageDump> {
+    return new Promise((resolve, reject) => {
+      this._getStorageTrie(address)
+        .then((trie) => {
+          const storage: StorageDump = {}
+          const stream = trie.createReadStream()
+
+          stream.on('data', (val: any) => {
+            storage['0x' + val.key.toString('hex')] = '0x' + val.value.toString('hex')
+          })
+          stream.on('end', () => {
+            resolve(storage)
+          })
+        })
+        .catch((e) => {
+          reject(e)
+        })
+    })
+  }
+
+  /**
+   * Checks if an `account` exists at `address`
+   * @param address - Address of the `account` to check
+   */
+  async accountExists(address: Address): Promise<boolean> {
+    log(`Verify if ${address.toString()} exists`)
+    const account = this._cache.get(address)
+    if (account.isEmpty()) {
+      // Get latest block (or block specified in `this.blockTag`)
+      const block = await this.provider.send('eth_getBlockByNumber', [this.blockTag, false])
+      // Get merkle proof for `address` from provider
+      const proof = await this.provider.send('eth_getProof', [address.toString(), [], block.number])
+      const proofBuf = proof.accountProof.map((proofNode: string) => toBuffer(proofNode))
+      const trie = new Trie({ useKeyHashing: true })
+      const verified = await trie.verifyProof(toBuffer(block.stateRoot), address.buf, proofBuf)
```

Comment:
@jochem-brouwer @g11tech  Do we need the block's `stateRoot` here or can we if we have already requested the proof can we just hash the first node and use that as the state root in the `verifyProof` call?  Seems like the latter should be correct, right?

Reply:
yes, that should give you the stateroot if you don't want to verify/double check